content/docs/ckan/harvester/_index.md

```diff
@@ -29,10 +29,10 @@ In the interface three methods are of particular interest and defining 80% of th
 * `fetch_stage` - checks what was saved with `gather_stage` and if information is not complete, goes back to source to fetch additional data.
 * `import_stage` - maps data to CKAN fields. This is the place to implement hooks for data preprocessing etc.
 
-Mapping logic is the same as in [profiles](https://github.com/ckan/ckanext-dcat/blob/master/ckanext/dcat/profiles.py): there is a base profile, describing fields mapping and extensions. FairDataPoint in this regard is an extension of [EuropeanDCATAP2Profile](https://github.com/ckan/ckanext-dcat/blob/1109205069dd105dda27e3486898e4ca1525a808/ckanext/dcat/profiles.py#L1487).
+<!-- Mapping logic is the same as in [profiles](https://github.com/ckan/ckanext-dcat/blob/master/ckanext/dcat/profiles.py): there is a base profile, describing fields mapping and extensions. FairDataPoint in this regard is an extension of [EuropeanDCATAP3Profile](https://github.com/ckan/ckanext-dcat/blob/1109205069dd105dda27e3486898e4ca1525a808/ckanext/dcat/profiles.py#L1487).
 
 If a dataset is extended with some fields on source side, depending on profile chosen, it is possible to configure that all the extra fields will be captured in an “extra“ blob with key-value pairs defined in the source (in `package_extra` DB table where key is always a string and value - is an object, flattened to string). A down side of it, if you interact with the dataset via UI then, all the fields which are not in the form will be ignored and then deleted because an update acts as “put“ operation, rather than a patch. Otherwise those field will appear one we map them properly.
 
 To be aware and to care about while doing mapping: if a field in a source has a name which is not mapped and you decide to map it later and assign another name, both alias will appear in CKAN.
 
-Another thing to bear in mind is that the validation is always runs on a full object. So on source schema update you may face a situation when a previously validated field for a dataset starts failing validation. Additionally, as validation runs only on a trigger, you need to run a script to migrate existing datasets on schema updates, otherwise you end up with a bunch of invalid datasets. Such a script can be written in python with [ckanapi](https://github.com/ckan/ckanapi/blob/master/README.md#remoteckan).
```

Comment:
@hcvdwerf suggestion: could you add somewhere here that harvesting is a slow process? I am testing locally and the Luxembourg FDP service, that contains only 6 datasets, takes 1.5 hours to finish.

I believe we could advise sysadmin to run it daily or in bigger intervals, depending on how big is the catalogue to be harvested.

Reply:
@brunopacheco1  Harvesting can be really fast, the main thing slowing it down is the number of distributions. IIRC the GDI datasets have a lot of those, while I can harvest ~200 datasets in a minute or two (but they don't or barely have distributions).

We might want to stick a profiler to it cus it is kinda ridiculous how slow it can be.

Reply:
Even if you add 300 distributions to each dataset coming from Lux FDP, 1500 entries is a tiny collection. Definetally there is an issue somewhere, algorithmical, architectural or environmental - probably a mix of all in some degree.

But indeed, profiling is an excellent idea to find the bottlenecks. Have you got experience on it? This monday I will support our ops team to troobleshoot our environment. After lest fix on ckan-docker, the harvesters partially work.

Reply:
Limited experience with it, but some willpower that can make up for it. You can always run the harvester-test for commandline and do poor mans stacktrace profiling (ctrl-c a bunch of times in the middle of harvesting, see which functions pop up the most).  But we can also try running the harvester with cProfile.